packages/components/src/components/text-input/_mixins.scss

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright IBM Corp. 2016, 2018
+//
+// This source code is licensed under the Apache-2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+@import '../../globals/scss/vars';
+@import '../../globals/scss/css--reset';
+@import '../../globals/scss/helper-mixins';
+@import '../../globals/scss/typography';
+@import '../../globals/scss/tooltip';
+@import '../form/form';
+@import './variables';
+
+/// Text input styles
+/// @access private
+/// @group text-input
+@mixin text-input {
+  $text-input-icon-size: carbon--type-icon-size(
+    map-get(map-get($tokens, 'body-short-01'), 'font-size')
+  );
+
+  .#{$prefix}--text-input {
+    @include reset;
+    @include type-style('body-short-01');
+    @include focus-outline('reset');
+    background-color: $field-01;
+    width: 100%;
+    height: $text-input-height;
+    padding: 0 $carbon--spacing-05;
+    color: $text-01;
+    border: none;
+    border-bottom: 1px solid $ui-04;
+    transition: background-color $duration--fast-01 motion(standard, productive),
+      outline $duration--fast-01 motion(standard, productive);
+
+    &:focus,
+    &:active {
+      @include focus-outline('outline');
+    }
+
+    &-wrapper svg[hidden] {
+      display: none;
+    }
+  }
+
+  .#{$prefix}--password-input {
+    padding-right: rem(40px);
+  }
+
+  .#{$prefix}--text-input::-webkit-input-placeholder {
+    @include placeholder-colors;
+  }
+
+  .#{$prefix}--text-input--light {
+    background-color: $field-02;
+  }
+
+  //-----------------------------
+  // Disabled & Error icon spacing
+  //-----------------------------
+  .#{$prefix}--text-input__field-wrapper {
+    position: relative;
+    display: flex;
+    align-items: center;
+    width: 100%;
+
+    .#{$prefix}--text-input__invalid-icon {
+      position: absolute;
+      right: rem(16px);
+      fill: $support-01;
+      height: $text-input-icon-size;
+      width: $text-input-icon-size;
```

Comment:
Maybe? It might not be that simple here? When I tried `height` and `width` last night it didn't work. Maybe something wrong the SVG's `preserveAspectRatio`? Just guessing at this point since I haven't had time to figure it out yet.